plugins/kotlin/uast/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/internal/firKotlinInternalUastUtils.kt

```diff
@@ -42,6 +43,27 @@ internal fun KtAnalysisSession.toPsiMethod(ktCall: KtCall): PsiMethod? {
     }
 }
 
+internal fun KtAnalysisSession.toPsiType(ktType: KtType, context: KtElement): PsiType {
+    if (ktType is KtNonErrorClassType && ktType.typeArguments.isEmpty()) {
+        // TODO: unclear when we need boxed type.
```

Comment:
@xiexed could you please help here?

Reply:
But the point is, I didn't see any places/outputs where boxed primitives are used even for FE 1.0 UAST.

Reply:
There are some usages of https://github.com/JetBrains/intellij-community/blob/kt-212-master/plugins/kotlin/uast/uast-kotlin/src/org/jetbrains/uast/kotlin/internal/kotlinInternalUastUtils.kt#L107 which has `boxed = true`. Type arguments and nullable types are always boxed, which seems quite reasonable.

Reply:
I see; I noticed that I've ignored that bit while commonizing many other expression kinds. Revisited; added a flag to the base service; and updated API use places accordingly if boxed primitives are required, such as receivers and type arguments.